File: backend/app/schemas/analyzer.py
from pydantic import BaseModel


class AllocationItem(BaseModel):
    symbol: str
    asset_type: str
    market_value: float
    weight: float  # 0–1


class PositionSummary(BaseModel):
    position_id: int
    symbol: str
    asset_type: str
    quantity: float
    avg_cost: float
    current_price: float
    market_value: float
    cost_basis: float
    unrealized_pnl: float
    unrealized_pnl_pct: float


class PortfolioSummary(BaseModel):
    total_market_value: float
    total_cost_basis: float
    total_unrealized_pnl: float
    total_unrealized_pnl_pct: float
    position_count: int
    positions: list[PositionSummary]
    allocation: list[AllocationItem]


class PricePoint(BaseModel):
    date: str
    open: float
    high: float
    low: float
    close: float
    volume: int


class RiskMetrics(BaseModel):
    symbol: str
    period: str
    sharpe_ratio: float
    max_drawdown: float
    cagr: float
    volatility: float


class PerformanceResponse(BaseModel):
    symbol: str
    period: str
    prices: list[PricePoint]
    metrics: RiskMetrics

File: frontend/src/types/analyzer.ts
export interface PositionSummary {
  position_id: number;
  symbol: string;
  asset_type: string;
  quantity: number;
  avg_cost: number;
  current_price: number;
  market_value: number;
  cost_basis: number;
  unrealized_pnl: number;
  unrealized_pnl_pct: number;
}

export interface AllocationItem {
  symbol: string;
  asset_type: string;
  market_value: number;
  weight: number;
}

export interface PortfolioSummary {
  total_market_value: number;
  total_cost_basis: number;
  total_unrealized_pnl: number;
  total_unrealized_pnl_pct: number;
  position_count: number;
  positions: PositionSummary[];
  allocation: AllocationItem[];
}

export interface PricePoint {
  date: string;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
}

export interface RiskMetrics {
  symbol: string;
  period: string;
  sharpe_ratio: number;
  max_drawdown: number;
  cagr: number;
  volatility: number;
}

export interface PerformanceResponse {
  symbol: string;
  period: string;
  prices: PricePoint[];
  metrics: RiskMetrics;
}
